src/coreclr/jit/flowgraph.cpp

```diff
@@ -6787,6 +6894,398 @@ void FlowGraphDominanceFrontiers::Dump()
 }
 #endif
 
+//------------------------------------------------------------------------
+// FlowGraphPostDominatorTree::IntersectPostdom:
+//   Find common IPDom parent, much like least common ancestor.
+//
+// Parameters:
+//   finger1 - A basic block that might share IPDom ancestor with finger2.
+//   finger2 - A basic block that might share IPDom ancestor with finger1.
+//
+// Returns:
+//   A basic block whose IPDom is the postdominator for finger1 and finger2, or else
+//   nullptr. This may be called while immediate postdominators are being computed,
+//   and if the input values are members of the same loop (each reachable from
+//   the other), then one may not yet have its immediate postdominator computed
+//   when we are attempting to find the immediate postdominator of the other. So a
+//   nullptr return value means that the the two inputs are in a cycle, not
+//   that they don't have a common postdominator ancestor.
+//
+// Remarks:
+//   See "A simple, fast dominance algorithm" by Keith D. Cooper, Timothy J.
+//   Harvey, Ken Kennedy.
+//
+BasicBlock* FlowGraphPostDominatorTree::IntersectPostdom(BasicBlock* finger1, BasicBlock* finger2)
+{
+    while (finger1 != finger2)
+    {
+        if ((finger1 == nullptr) || (finger2 == nullptr))
+        {
+            return nullptr;
+        }
+        while ((finger1 != nullptr) && (finger1->bbReversePostorderNum < finger2->bbReversePostorderNum))
+        {
+            finger1 = finger1->bbIPDom;
+        }
+        if (finger1 == nullptr)
+        {
+            return nullptr;
+        }
+        while ((finger2 != nullptr) && (finger2->bbReversePostorderNum < finger1->bbReversePostorderNum))
+        {
+            finger2 = finger2->bbIPDom;
+        }
+    }
+    return finger1;
+}
```

Comment:
Are the null checks needed? They were removed from `FlowGraphDominatorTree::IntersectDom` in #103803.